notesapi/v1/models.py

```diff
@@ -0,0 +1,141 @@
+import json
+from django.db import models
+from django.core.exceptions import ValidationError
+from django.conf import settings
+from django.db.models import signals
+from django.dispatch import receiver
+from elasticutils.contrib.django import Indexable, MappingType
+
+
+class Note(models.Model):
+    """
+    Annotation model.
+    """
+    user_id = models.CharField(max_length=255)
+    course_id = models.CharField(max_length=255)
+    usage_id = models.CharField(max_length=255)
+    text = models.TextField(default="")
+    quote = models.TextField(default="")
+    ranges = models.TextField(default="")
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+
+    def clean(self, note):
```

Comment:
> this deviates from the Model.clean definition (which takes no arguments and operates on instance data in-place) - I think it would be best to override the classmethod Model.create to populate the model from dict values, and then use Model.clean only if needed to provide custom validation that the default implementation doesn't provide.

Good idea! Added `Model.create`.